src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -285,6 +296,8 @@ public virtual VerifyResult VerifyStateDependent(StoreView snapshot, Transaction
         {
             if (ValidUntilBlock <= snapshot.Height || ValidUntilBlock > snapshot.Height + MaxValidUntilBlockIncrement)
                 return VerifyResult.Expired;
+            if (snapshot.GetBlock(LastBlockHash)?.Index != ValidUntilBlock - MaxValidUntilBlockIncrement)
```

Comment:
@Qiao-Jin, shouldn't it be `<=` instead of `!=`? If not, could you, please, explain the meaning of this line?

Reply:
Why? This `LastBlockHash` is used to prove whether it's a valid ValidUntilBlock got from Height + MaxValidUntilBlockIncrement.